app/controllers/api/v1/stats/channels_controller.rb

```diff
@@ -5,24 +5,15 @@ def index
 
   def show
     channel = Channel.find(params[:id])
+    channel_details = channel.details
 
     case channel.details_type
-
       when "SiteChannelDetails"
         channel_details = SiteChannelDetails.find_by_id(channel.details_id)
         channel_type = "website"
         channel_name = channel_details.url
-      when "YoutubeChannelDetails"
-        channel_details = YoutubeChannelDetails.find_by_id(channel.details_id)
-        channel_type = "youtube"
-        channel_name = channel_details.name
-      when "TwitterChannelDetails"
-        channel_details = TwitterChannelDetails.find_by_id(channel.details_id)
-        channel_type = "twitter"
-        channel_name = channel_details.name
-      when "TwitchChannelDetails"
-        channel_details = TwitchChannelDetails.find_by_id(channel.details_id)
-        channel_type = "twitch"
+      else
+        channel_type = channel.details_type.gsub("ChannelDetails", "").downcase
```

Comment:
I think I created a function for something like this way back when, check out https://github.com/brave-intl/publishers/blob/a5918610172eb28fea0c71e45e2438d9f4f0c87c/app/models/channel.rb#L306-L308

Reply:
Although `type_display` probably wasn't the best name for it, feel free to rename (just search for any instances if you do)